### PR TITLE
Fix hanging paragraphs for C++20

### PR DESCRIPTION
**Pull request by zygoloid**
Mechanically fix all hanging paragraphs by adding "General" sections.

For C++23, we should go through these and rename them as appropriate, but for C++20 let's just mechanically call them all "General" and name them [foo.general].

Fixes cplusplus/nbballot#394.

### CODE REVIEW

File: source/declarations.tex [hunk 1]
```diff
@@ -2084,6 +2090,8 @@
 \indextext{specifier|)}%
 
 \rSec1[dcl.decl]{Declarators}%
+
+\rSec2[dcl.decl.general]{General}%
 \indextext{declarator|(}
```
Comment (jensmaurer) [inline source/declarations.tex:2095]: This is not nice; the "declarator" start of index range should probably be before the "General" section. But maybe this is not the right time to fiddle with this.

**Review by jensmaurer — APPROVED** [review]
I'm creating an extension for tools/check.sh to detect hanging paragraphs using sed and awk.

**Comment by johelegp** [comment]
Fixes #3525 (I think).
Partially addresses #1771.
Related to #1730 and #574.

File: source/declarations.tex (hunk 1 above)
Comment (zygoloid) [reply source/declarations.tex:2095]: We should fix this for the `master` branch. I think this is fine for C++20.

**Comment by zygoloid** [comment]
> Fixes #3525 (I think).

Well, technically this PR doesn't fix #3525, but the merge to master will :)

**Comment by zygoloid** [comment]
> I'm creating an extension for tools/check.sh to detect hanging paragraphs using sed and awk.

Ah, we were both working on this at the same time. I built a perl script to do this; do you prefer one approach or the other?

**Comment by jensmaurer** [comment]
My check script is here: #4170

**Comment by johelegp** [comment]
> > Fixes #3525 (I think).
> 
> Well, technically this PR doesn't fix #3525, but the merge to master will :)

Maybe not:
> # FIXME: We should also check for subclauses that have no siblings.

Does your script fixes this, @jensmaurer?

**Comment by jensmaurer** [comment]
@zygoloid , your script introduces a dependency on perl, which is novel for the check.sh script. My variant just uses sed and perl.

@johelegp , no, my script does not check for lone siblings. I think those only currently appear together with hanging paragraphs.

**Comment by johelegp** [comment]
I've marked as done those that will be fixed by this. Please, do check the others: #3525.

**Comment by zygoloid** [comment]
> @zygoloid , your script introduces a dependency on perl, which is novel for the check.sh script. My variant just uses sed and awk. 

For what it's worth, using `perl` here is actually more portable than your uses of `awk` (which rely on GNU awk-isms, and consequently don't work out of the box on Mac OS). But I'm fine sticking with something more similar to what we already do here if there's a worry that `perl` won't be available. Dropping the checking part from this PR.

**Comment by jensmaurer** [comment]
@zygoloid , I think I'm not using GNU awk specifics, at least not in the "hanging paragraph" check.
(I think you discovered a GNU sed dependency, though.)

**Comment by zygoloid** [comment]
Sorry, you're right, it's `sed` not `awk` that's the problem in this script in general.

**Comment by zygoloid** [comment]
See also follow-up fixes #4173 and #4174.

**Pull request merged**